var/spack/repos/builtin/packages/bazel/package.py

```diff
@@ -120,6 +122,39 @@ class Bazel(Package):
 
     executables = ["^bazel$"]
 
+    # Download resources to perform offline build with bazel.
+    # The following URLs and sha256 are in the file distdir_deps.bzl at the root of bazel sources.
+    resource_dictionary = {}
+    resource_dictionary["bazel_skylib"] = {
+        "url": "https://github.com/bazelbuild/bazel-skylib/releases/download/1.0.1/bazel-skylib-1.0.1.tar.gz",
+        "sha256": "f1c8360c01fcf276778d3519394805dc2a71a64274a3a0908bc9edff7b5aebc8",
+        "when": "@4:6",
+    }
+    resource_dictionary["zulu_11_56_19"] = {
+        "url": "https://mirror.bazel.build/cdn.azul.com/zulu/bin/zulu11.56.19-ca-jdk11.0.15-linux_x64.tar.gz",
+        "sha256": "e064b61d93304012351242bf0823c6a2e41d9e28add7ea7f05378b7243d34247",
+        "when": "@6",
+    }
+    resource_dictionary["zulu_11_50_19"] = {
+        "url": "https://mirror.bazel.build/openjdk/azul-zulu11.50.19-ca-jdk11.0.12/zulu11.50.19-ca-jdk11.0.12-linux_x64.tar.gz",
+        "sha256": "b8e8a63b79bc312aa90f3558edbea59e71495ef1a9c340e38900dd28a1c579f3",
+        "when": "@5",
+    }
```

Comment:
It looks like `azul-jdk` is its own implementation of Java: I should say I think this is seems to conflict somewhat with our dependency mechanism (not in the sense that it should cause an immediate failure, but in the sense that it subverts our tracking of how things are built): `bazel` declares a dependency on `java` which means that before we install `bazel` we install an implementation of the `jdk`.

I lack the context to say whether this is wrong for this package in particular (IIRC there are some other aspects of `bazel` that don't strictly fit into our idioms).

Reply:
You're right, this PR suggests doing manually what Bazel did automatically anyway when building it.
This will allow, initially, to carry out the offline installation of bazel which is currently blocked in centers which do not have access to the internet. Then we can focus on cleaning up dependencies such as Java.